common/evm_block_ref.go

```diff
@@ -101,6 +97,54 @@ func ExtractEvmBlockReferenceFromRequest(r *JsonRpcRequest) (string, int64, erro
 			return "", 0, fmt.Errorf("unexpected missing 2nd parameter for method %s: %+v", r.Method, r.Params)
 		}
 
+	case "eth_getBalance",
+		"eth_getTransactionCount",
+		"eth_getCode",
+		"eth_call":
+		if len(r.Params) > 1 {
+			switch secondParam := r.Params[1].(type) {
+			case string:
+				// Handle the second parameter as a HEX String
+				if strings.HasPrefix(secondParam, "0x") {
```

Comment:
Do we need to consider block hash as well here?

e.g.:
```suggestion
				if len(secondParam) < 10 && strings.HasPrefix(secondParam, "0x") {
```

Or even further extend to return blockHash if len is 66, etc

Reply:
if I understood correctly from (https://eips.ethereum.org/EIPS/eip-1898), the string can also be one of the below values (prob for backward. compatibility):

* HEX String - an integer block number
* String “earliest” for the earliest/genesis block
* String “latest” - for the latest canonical block
* String “pending” - for the pending state/transactions
* String “safe” - for the most recent safe block
* String “finalized” - for the most recent finalized block

Seems like `blockHash` is only supported in as part of the object, hence making if the param is either a reference to Data (`blockHash`) or Quantity (`blockNumber`)

Reply:
I think it's better to include this logic, to be robust against other client behavior, since it won't impact the existing expected flow and won't add too much cost.

Basically assume some unknown client/chain might still allow blockHash here.